### ISSUE TITLE
Vscode suggestions and autocompletion is not working

### ISSUE THREAD

**Issue by esimov** [golang/vscode-go#2324]
### What version of Go, VS Code & VS Code Go extension are you using?
<details><summary>Version Information</summary><br>

* Run `go version` to get version of Go from _the VS Code integrated terminal_.
	- go version go1.18.3 linux/amd64
* Run `gopls -v version` to get version of Gopls from _the VS Code integrated terminal_.
	- golang.org/x/tools/gopls v0.9.0
* Run `code -v` or `code-insiders -v` to get version of VS Code or VS Code Insiders.
	- 1.69.0
```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/esimo/.cache/go-build"
GOENV="/home/esimo/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/esimo/projects/Go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/esimo/projects/Go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18.3"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/esimo/projects/Go/src/esimov/go-generics/go.mod"
GOWORK="/home/esimo/projects/Go/src/esimov/go.work"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build1041229074=/tmp/go-build -gno-record-gcc-switches"
```
</details>

### Share the Go related settings you have added/edited

```
"go.toolsManagement.autoUpdate": true,
"go.alternateTools": {},
"go.formatTool":"goimports",
"go.useLanguageServer": true,
"[go]": {
    "editor.tabSize": 8,
    "editor.insertSpaces": false,
    "editor.detectIndentation": true
}
```
### Describe the bug
I'm working on a WSL environment and Vscode suddenly stopped showing the suggestions and the code autocompletion functionality has stopped working too. I've tried reinstalling the Go tools by pressing `CTRL+Shift+P` and selecting `Go: Install/Update Tools`. I ran also the `go mod tidy` command to update the required packages. Another attempt was to execute `go work use -r ./`, because this helped in this case #2259, but didn't helped here. Instead I was getting something like in the snapshot below:

<img width="319" alt="Screenshot 2022-07-08 130438" src="https://user-images.githubusercontent.com/883386/177986543-f3abcc84-4088-4b53-b06c-24460d167b6b.png">

The package autoimport also is not working anymore. I'm out of ideas what else could I try.

**Comment by findleyr** [comment]
Hi, could you try restarting gopls and collecting logs, as described here?
https://github.com/golang/vscode-go/blob/master/docs/troubleshooting.md#restart-gopls

**Comment by esimov** [comment]
@wtask no, it's not that case. Even if I'm changing the variable name to something else, it's still not working. BTW the autocompletion and suggestion is not working on other workspaces/projects of mine.

**Comment by esimov** [comment]
@findleyr I have tried also to restart gopls, but it's still not working.

**Comment by findleyr** [comment]
@esimov can you plesae try capturing logs? That will help us diagnose the problem.

**Comment by wtask** [comment]
> @wtask no, it's not that case. Even if I'm changing the variable name to something else, it's still not working. BTW the autocompletion and suggestion is not working on other workspaces/projects of mine.

Do you save the source after renaming?

**Comment by esimov** [comment]
It's not needed to save it after renaming, it should work instantly on typing, at least it worked that way until I have encountered this problem.

**Comment by findleyr** [comment]
Correct, saving shouldn't matter in this case.

The likely scenario is that you were broken by the upgrade to gopls@v0.9.0, but before downgrading I would like to capture logs so that we can understand the failure.

**Comment by esimov** [comment]
These are the errors I'm getting:

```bash
[Error - 4:48:17 PM] 2022/07/08 16:48:17 imports fixes: AllImportsFixes: found module "github.com/gorilla/websocket" twice in the workspace
	file="/home/.../cache.go"

[Error - 4:48:17 PM] 2022/07/08 16:48:17 imports fixes: AllImportsFixes: found module "github.com/gorilla/websocket" twice in the workspace
	file="/home/.../range_test.go"

[Error - 4:48:17 PM] 2022/07/08 16:48:17 imports fixes: AllImportsFixes: found module "github.com/gorilla/websocket" twice in the workspace
	file="/home/.../range_test.go"
```

I'm not sure why gorilla websocket is considered to be interfered with my module, since I don't use it at all. I'm using in another module, but as I mentioned I didn't had this problem before updating to the latest vscode version, so it' might be related to an issue in the latest release.

**Comment by findleyr** [comment]
Thank you for that. This is helpful.

Are you using either a `go.work` file, or the `experimentalWorkspaceModule` setting?

**Comment by esimov** [comment]
Yes, I do have a `go.work` file and in the root of my projects directory. No, I'm not using `experimentalWorkspaceModul` setting and I don't even know what is it.

**Comment by findleyr** [comment]
Got it. I believe I have found the bug. We are failing on some old logic for the `experimentalWorkspaceModule` setting even though it is not active. You shouldn't need to know what that settings is: it is deprecated in favor of `go.work`.

Can you try removing all but one module from your `go.work`, and confirm whether this makes the problem go away when working on that one module?

**Comment by esimov** [comment]
I have already tried it, but the issue didn't went away. As it might help you in identifying the problem the gorilla websocket module is saved into the vendor folder, so it's not imported directly, it's loaded from the vendor folder, so this package is showing in two places, but in different vendor folders.

```
./project1/vendor/github.com/gorilla/websocket
./project2/vendor/github.com/gorilla/websocket
```

Note: I have replaced the actual project names.

**Comment by esimov** [comment]
Are there any updates about the issue?

**Comment by dzpt** [comment]
I have the same issue, no auto suggestion anymore.
golang.org/x/tools/gopls v0.9.1
go1.18.4 darwin/amd64
gotta jump back to the previous version.

**Comment by findleyr** [comment]
This is a gopls regression that we will fix for v0.9.2, scheduled for early August (though once it is fixed you can of course rebuild gopls from master).

However, I am not able to reproduce with a `go.work` file. @esimov it should not be possible to hit this error if you have a `go.work` file using `project1` and `project2`.  @esimov where is your `go.work` file? Is it in a parent directory of `./parent1` and `./parent2`?

@dzpt thank you for the signal boost, and for sharing that this does not affect v0.8.4.

Transferring to the gopls issue tracker.
